### PR TITLE
[MB-7411] Display service item calculations when reviewing a payment request

### PR DESCRIPTION
**Pull request by duncan-truss**
## Description

This PR displays the payment service item pricing calculations on payment requested still in the Needs Review status.  There is a button that will toggle the display of the pricing component.  The toggle is now added to the payment requests in the Reviewed status as well.

## Reviewer Notes

Most of the tests for the ServiceItemCard.jsx from before are in ReviewServiceItems.test.jsx but I created a new ServiceItemCard.test.jsx to house the new logic.

I thought about combining the ServiceItemCard storybook stories into a single one with controls for the shipment type but I thought it would be easier to catch if something was off in a Happo diff if they stayed how they were.  It's mostly just visual so I probably could have done it.

We don't yet have calculations for basic service items but I tried not to add a ton of workaround logic even though they have some params they won't get displayed.

I disabled some of the scroll logic that started to be more jarring than helpful on non-basic service items since they will never appear 2, to a page.

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make db_dev_e2e_populate
make server_run
make office_client_run
```

1. Login as the TIO office user
2. Select the payment request with the move code `PARAMS`
3. Click on the button to review service items of the payment request
4. When on a service item that has calculations (not basic service items) click on the button "Show calculations" to reveal the service item pricing calculations.  The text of the button should now read "Hide calculations" and clicking again will hide the pricing calculations.
5. Approve or reject all the service items in a payment request to change the status to REVIEWED.
6. Click on the `View documents` to review the payment service items again.
7. The "Show calculations" button and pricing calculations should behave as before. 

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely) have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-7411) for this change

## Screenshots

Screen Shot 2021-04-13 at 5 45 52 PM
Screen Shot 2021-04-13 at 5 45 43 PM
Screen Shot 2021-04-13 at 5 44 06 PM
Screen Shot 2021-04-13 at 5 43 57 PM
Screen Shot 2021-04-13 at 5 43 45 PM

### CODE REVIEW

**Comment by Ryan-Koch** [comment]
One thing I saw is that it looks like we're showing the show calculations link for basic service items, which I think won't have any params. We might want to consider not rendering the link for those two since it won't expand any new information. That said, if it's a lot to do we can always peel that off into a follow up task.

**Comment by amanda5iu** [comment]
When I click `Reject` then `Show calculations`, it appears to shorten the width of the entire sidebar, which squeezes the text label on some of the calculations (see `Total amount requested`). It doesn't happen when `Accept` is selected though. 

Screen Shot 2021-04-14 at 12 50 47 PM

**Comment by duncan-truss** [comment]
> When I click `Reject` then `Show calculations`, it appears to shorten the width of the entire sidebar, which squeezes the text label on some of the calculations (see `Total amount requested`). It doesn't happen when `Accept` is selected though.
> 
> Screen Shot 2021-04-14 at 12 50 47 PM

@amanda5iu I'm not sure I'm seeing the width change, could you tell me what browser and service item you are looking at? You did identify another bug though where we weren't attaching the parms after the status was updated ❤️!

**Comment by duncan-truss** [comment]
> One thing I saw is that it looks like we're showing the show calculations link for basic service items, which I think won't have any params. We might want to consider not rendering the link for those two since it won't expand any new information. That said, if it's a lot to do we can always peel that off into a follow up task.

@Ryan-Koch I made the change to not show the button if it's an unsupported service item such as the basic ones ✅

**Review by donaldthai — APPROVED** [review]
LGTM

File: src/components/Office/ReviewServiceItems/ServiceItemCard.jsx [hunk 1]
```diff
@@ -42,16 +45,34 @@ const ServiceItemCard = ({
...
           </dl>
-
-          {paymentServiceItemParams.length > 0 && (
-            <ServiceItemCalculations
-              totalAmountRequested={amount * 100}
-              serviceItemParams={paymentServiceItemParams}
-              itemCode={mtoServiceItemCode}
-              tableSize="small"
-            />
+          {allowedServiceItemCalculations.includes(mtoServiceItemCode) && paymentServiceItemParams.length > 0 && (
```
Comment (YanZ777) [inline src/components/Office/ReviewServiceItems/ServiceItemCard.jsx:48]: Correct me if I'm wrong, but this chunk of code looks the same as 134 - 161. Would it be worthwhile to extract this out to its own component?

Comment (duncan-truss) [reply src/components/Office/ReviewServiceItems/ServiceItemCard.jsx:48]: I agree it is pretty much the same.  I'm not sure the button with the toggle is enough for a standalone component but maybe I can define it just once prior to the render as a single variable.  Most of the conditionals will go away when all of the service item pricing calculations are implemented.

**Review by Ryan-Koch — APPROVED** [review]
Nicely done!

**Review by YanZ777 — APPROVED** [review]

**Review by amanda5iu — APPROVED** [review]
The calculations stopped shrinking :)

**Pull request merged**